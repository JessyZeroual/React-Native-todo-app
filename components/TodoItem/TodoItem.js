import React from "react";
import { Text } from "react-native";

import CheckBox from "../../utils/CheckBox";
import SwipeRow from "../../utils/SwipeRow";
import truncateString from "../../utils/truncateString";
import { TodoItemWrapper, Title, Logo } from "./TodoItem.styled";

const TodoItem = ({ item }) => {
  return (
    <SwipeRow
      key={item.key}
      item={item}
      swipeThreshold={-150}
      onSwipe={() => console.log(`delete ${item.title}`)}
    >
      <TodoItemWrapper>
        <CheckBox completed={item.completed} />
        <Title completed={item.completed}>
          {truncateString(item.title, 25)}
        </Title>
        <Logo
          source={{
            uri: "https://reactnative.dev/img/tiny_logo.png",
          }}
        />
      </TodoItemWrapper>
    </SwipeRow>
  );
};

export default TodoItem;
