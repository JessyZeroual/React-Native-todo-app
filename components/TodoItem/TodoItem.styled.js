import styled from "styled-components/native";
import { primaryColor } from "../../constant/style-constants";
import Animated from "react-native-reanimated";

export const TodoItemWrapper = styled(Animated.View)`
  display: flex;
  align-items: center;
  flex-direction: row;
  background: #fff;
  height: 80px;
  margin: 10px;
  border-radius: 5px;
  padding: 0px 20px;
`;

export const Title = styled.Text`
  text-decoration: ${({ completed }) => completed && "line-through"};
`;

export const Logo = styled.Image`
  margin-left: auto;
  height: 50px;
  width: 50px;
`;
