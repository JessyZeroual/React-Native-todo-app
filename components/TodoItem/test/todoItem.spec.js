import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import TodoItem from "../TodoItem";

jest.mock("../../../utils/truncateString");

describe("TodoItem", () => {
  let renderer;

  const mockDataItem = {
    id: "id-1",
    name: "Todo-1",
  };

  beforeEach(() => {
    act(() => {
      renderer = TestRenderer.create(<TodoItem item={mockDataItem} />);
    });
  });

  describe("When TodoList is mounted", () => {
    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("has item in pros", () => {
      expect(renderer.root.props.item).toEqual(mockDataItem);
    });
  });
});
