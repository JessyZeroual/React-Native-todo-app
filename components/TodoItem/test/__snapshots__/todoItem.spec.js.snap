// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem When TodoList is mounted renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#dcdcdc",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "row",
        "height": 80,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactnative.dev/img/tiny_logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 50,
          "marginBottom": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 0,
          "width": 50,
        },
      ]
    }
  />
  <Text />
</View>
`;
